File: Kata 7/XsAndOs/XsAndOs.py
"""
Kata Level: 7.

DESCRIPTION:
Check to see if a string has the same amount of 'x's and 'o's. 
The method must return a boolean and be case insensitive. The string can contain any char.

Examples input/output:

XO("ooxx") => true
XO("xooxx") => false
XO("ooxXm") => true
XO("zpzpzpp") => true // when no 'x' and 'o' is present should return true
XO("zzoo") => false
"""

# Mi solución en Python.
	
def xo(s):
    contador = 0
    for letter in s:
        if letter == 'x' or letter == 'X':
            contador += 1
        elif letter == 'o' or letter == 'O':
            contador -= 1
    return contador == 0

# Solución de la comunidad 1.

def xo_community(s):
    s = s.lower()
    return s.count('x') == s.count('o')

print(xo("ooxx"))
print(xo("xooxx"))
print(xo("ooxXm"))
print(xo("zpzpzpp"))
print(xo("zzoo"))

File: Kata 7/XsAndOs/XsAndOs.cs
/* Kata Level: 7.
 * 
 * DESCRIPTION:
 * Check to see if a string has the same amount of 'x's and 'o's. 
 * The method must return a boolean and be case insensitive. The string can contain any char.
 * 
 * Examples input/output:
 * 
 * XO("ooxx") => true
 * XO("xooxx") => false
 * XO("ooxXm") => true
 * XO("zpzpzpp") => true // when no 'x' and 'o' is present should return true
 * XO("zzoo") => false
 */

public class Kata
{
    public static bool XO (string input)
	{
		return input.Where(x => x == 'x' || x == 'X').Count() == input.Where(x => x == 'o' || x == 'O').Count();
	}
    public static void Main(string[] args)
    {
        System.Console.WriteLine(XO("ooxx"));
        System.Console.WriteLine(XO("xooxx"));
        System.Console.WriteLine(XO("ooxXm"));
        System.Console.WriteLine(XO("zpzpzpp"));
        System.Console.WriteLine(XO("zzoo"));
    }
}
	